bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/data/v2/wrappers/Filters.java

```diff
@@ -0,0 +1,587 @@
+      range.setFamilyName(Preconditions.checkNotNull(family));
+    }
+
+    public QualifierRangeFilter startClosed(ByteString value) {
+      range.setStartQualifierClosed(value);
+      return this;
+    }
+
+    public QualifierRangeFilter startOpen(ByteString value) {
+      range.setStartQualifierOpen(value);
+      return this;
+    }
+
+    public QualifierRangeFilter endClosed(ByteString value) {
+      range.setEndQualifierClosed(value);
+      return this;
+    }
+
+    public QualifierRangeFilter endOpen(ByteString value) {
+      range.setEndQualifierOpen(value);
+      return this;
+    }
+
+    @Override
+    public RowFilter toProto() {
+      return RowFilter.newBuilder().setColumnRangeFilter(range.build()).build();
+    }
+  }
+
+  public static class QualifierFilter {
+
+    /**
+     * Matches only cells from columns whose qualifiers satisfy the given <a
+     * href="https://github.com/google/re2/wiki/Syntax>RE2 regex</a>. Note that, since column
+     * qualifiers can contain arbitrary bytes, the `\C` escape sequence must be used if a true
+     * wildcard is desired. The `.` character will not match the new line character `\n`, which may
+     * be present in a binary qualifier.
+     */
+    public Filter regex(String regex) {
+      return regex(ByteString.copyFromUtf8(regex));
+    }
+
+    /**
+     * Matches only cells from columns whose qualifiers satisfy the given <a
+     * href="https://github.com/google/re2/wiki/Syntax>RE2 regex</a>. Note that, since column
+     * qualifiers can contain arbitrary bytes, the `\C` escape sequence must be used if a true
+     * wildcard is desired. The `.` character will not match the new line character `\n`, which may
+     * be present in a binary qualifier.
+     */
+    public Filter regex(ByteString regex) {
+      return new SimpleFilter(
+          RowFilter.newBuilder()
+              .setColumnQualifierRegexFilter(regex)
+              .build());
+    }
+
+    /** Matches only cells from columns whose qualifiers equal the value. */
+    public Filter exactMatch(ByteString value) {
+      return new SimpleFilter(
+          RowFilter.newBuilder()
+              .setColumnQualifierRegexFilter(RegexUtil.literalRegex(value))
+              .build());
+    }
+
+    /**
+     * Construct a {@link QualifierRangeFilter} that can create a {@link ColumnRange} oriented
+     * {@link Filter}.
+     * @return a new {@link QualifierRangeFilter}
+     */
+    public QualifierRangeFilter range(String family) {
+      return new QualifierRangeFilter(family);
+    }
+  }
+
+  /**
+   * Matches only cells with microsecond timestamps within the given range.  Start is inclusive
+   * and end is exclusive.
+   */
+
+  public static class TimestampRangeFilter extends Filter {
+    private final TimestampRange.Builder range = TimestampRange.newBuilder();
+
+    public TimestampRangeFilter startClosed(long startMicros) {
+      range.setStartTimestampMicros(startMicros);
```

Comment:
nit: consider handling Long.MIN_VALUE as a trigger to clearStartTimestampMicros (same for endOpen for Long.MAX_VALUE)